### PR TITLE
[YUNIKORN-2528] Increase coverage for UGM code

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
There are some code paths in UGM which are not covered by unit tests.  

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2528

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/ugm/queue_tracker.go [hunk 1]
```diff
@@ -403,24 +403,20 @@ func (qt *QueueTracker) canRunApp(hierarchy []string, applicationID string, trac
...
-		qt.maxRunningApps == 0 && resources.IsZero(qt.maxResources) {
-		return true
-	}
-	return false
+	// reached a leaf queue
+	return len(qt.runningApplications) == 0 &&
+		resources.IsZero(qt.resourceUsage) &&
+		len(qt.childQueueTrackers) == 0 &&
+		qt.maxRunningApps == 0 &&
+		resources.IsZero(qt.maxResources)
```
Comment (pbacsko) [inline pkg/scheduler/ugm/queue_tracker.go:421]: @manirajv06 please check this logic, I think this is simpler than the original and accomplishes the same.

Comment (manirajv06) [reply pkg/scheduler/ugm/queue_tracker.go:421]: In newer approach, we take decision only after reaching the leaf queue by traversing all the queues in the queue hierarchy. Whereas earlier, earlier one takes decision at very earlier stage and come out then and there.

Comment (pbacsko) [reply pkg/scheduler/ugm/queue_tracker.go:421]: Even in the old code, `canBeRemovedInternal()` only returns true if `len(qt.childQueueTrackers) == 0`. So you need to descend down the hierarchy. You can't return earlier, therefore the optimization is not really there.

Comment (manirajv06) [reply pkg/scheduler/ugm/queue_tracker.go:421]: For example,

root.parent running app is 10. root.parent has 100's of child queues (or child queue trackers) below it. In earlier approach, We can come out right away the moment we see 10 apps running in root.parent queue and confirm that it is better not to remove root.parent and any queue below it.

Comment (pbacsko) [reply pkg/scheduler/ugm/queue_tracker.go:421]: Ok I'm reverting this change, let's not spend too much time on this. Maybe I'll get back to it later.

**Review by manirajv06 — APPROVED** [review]
looks good.

Are there any more places to be covered? JIRA description talks about manager methods too.

**Pull request closed without merging**